docs: add example for enabling both AVIF and WebP image formats for better image optimization

KaziMahbuburRahman opened the pull request:
## Title
docs: add example for enabling both AVIF and WebP image formats

## Description

### What?
This PR enhances the image component documentation by adding examples showing how to enable both AVIF and WebP image formats together in `next.config.js`. It also updates the `image-component` example to demonstrate this configuration.

### Why?
Currently, the documentation only shows examples for:
- WebP only (default): `formats: ['image/webp']`
- AVIF only: `formats: ['image/avif']`

However, many developers want to enable both formats together to:
- Prefer AVIF for browsers that support it (better compression, ~20% smaller files)
- Fall back to WebP for broader browser compatibility
- Maximize image optimization across different browsers

This configuration is a best practice but wasn't clearly documented with an example. 

### How?
- Added example configuration `formats: ['image/avif', 'image/webp']` to App Router image component documentation
- Added the same example to Pages Router (legacy) image component documentation for consistency
- Updated `examples/image-component/next.config.js` to demonstrate the configuration
- Added clarification that the order matters - Next.js uses the first format in the array that the browser supports

### Changes Made
- `docs/01-app/03-api-reference/02-components/image.mdx` - Added AVIF + WebP example
- `docs/02-pages/04-api-reference/01-components/image-legacy.mdx` - Added AVIF + WebP example  
- `examples/image-component/next.config.js` - Added formats configuration

### Testing
- [x] Documentation changes only - no code changes
- [x] Verified examples are syntactically correct
- [x] Checked that formatting follows Next.js documentation guidelines

### Checklist
- [x] Run `pnpm prettier-fix` to fix formatting issues (if applicable)
- [x] Documentation follows the docs guidelines
- [x] Changes are clear and helpful for developers

Commit:
Enhance image component documentation to include support for AVIF and WebP formats. Added configuration examples and clarified caching behavior for multiple formats.

ijjk commented:
Allow CI Workflow Run

- [ ] approve CI run for commit: 6b5d5e6988d5ce2a6cf0e6f5b7acb0c30d1967b6

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

KaziMahbuburRahman committed:
Merge branch 'canary' into docs/add-avif-webp-formats-example

icyJoseph commented on docs/01-app/03-api-reference/02-components/image.mdx:
````diff
@@ -752,10 +752,21 @@ module.exports = {
+```js filename="next.config.js"
+module.exports = {
+  images: {
+    formats: ['image/avif', 'image/webp'],
````
Does this introduce more cache overhead though? I don't mean as in, don't do it cuz of the overhead, but as in, we should probably mention this somewhere

KaziMahbuburRahman replied:
Thanks for the feedback! I've added a note about cache overhead in the "Good to know" section. The documentation now mentions that when using multiple formats, Next.js will cache each format separately, which means increased storage requirements compared to using a single format, as both AVIF and WebP versions of images will be stored for different browser support.

This should help developers make an informed decision when choosing between single vs. multiple formats.

icyJoseph commented on docs/01-app/03-api-reference/02-components/image.mdx:
```diff
@@ -752,10 +752,21 @@ module.exports = {
 >
 > - We still recommend using WebP for most use cases.
 > - AVIF generally takes 50% longer to encode but it compresses 20% smaller compared to WebP. This means that the first time an image is requested, it will typically be slower, but subsequent requests that are cached will be faster.
+> - When using multiple formats, the order matters. Next.js will use the first format in the array that the browser supports.
```
Do you think this necessary? we do mention this in 755.

KaziMahbuburRahman replied:
Good catch! You're right - we already mention that "the array order matters" earlier in the documentation (line 743). I've removed the redundant bullet point about order mattering from the "Good to know" section to avoid duplication.

The earlier explanation at line 743 already covers this: "If the Accept header matches more than one of the configured formats, the first match in the array is used. Therefore, the array order matters."

Commit:
docs: address reviewer feedback on AVIF/WebP formats example

- Add note about cache overhead when using multiple formats
- Remove redundant note about array order (already mentioned earlier)

Addresses feedback about documenting increased storage requirements
when enabling both AVIF and WebP formats, and removes duplicate
information about format order.

Commit:
Merge branch 'docs/add-avif-webp-formats-example' of https://github.com/KaziMahbuburRahman/next.js into docs/add-avif-webp-formats-example

KaziMahbuburRahman committed:
Merge branch 'canary' into docs/add-avif-webp-formats-example

Commit:
Merge branch 'docs/add-avif-webp-formats-example' of https://github.com/KaziMahbuburRahman/next.js into docs/add-avif-webp-formats-example

KaziMahbuburRahman committed:
Merge branch 'canary' into docs/add-avif-webp-formats-example

KaziMahbuburRahman committed:
Merge branch 'canary' into docs/add-avif-webp-formats-example

icyJoseph commented on examples/image-component/next.config.js:
```diff
@@ -1,6 +1,7 @@
 /** @type {import('next').NextConfig} */
 module.exports = {
   images: {
+    formats: ['image/avif', 'image/webp'],
```
Maybe we can remove this one though? wdyt?

KaziMahbuburRahman replied:
Thanks for the suggestion! Since the docs update is specifically about showcasing the AVIF+WebP configuration, I’d prefer to keep the formats entry in this example so readers can run the sample project and immediately see the setup in action. It also keeps the example consistent with the guidance we just added to the docs.

icyJoseph replied:
I see, I was thinking more in terms of what we are teaching here. 

Even with all of AVIF's advantages, even if there's 1 user browsing with only Webp support, in the worst case scenario, we'd duplicate storage. And as far as I understand, compression w/ AVIF takes longer, so more CPU time is used. 

So, for some projects, they'll have more CPU time on image compression, and double storage, assuming worst case of course.

Again this is in the learn/teaching area, but it is important to consider. For example in Next 16 we increased the minimumCacheTTL, to 4 hours, to reduce CPU usage, for images that do not have a Cache Control header.

Because with Webp, the compression is still not as good, you do end up storing a bit more than w/ only AVIF, and it follows that your images download slower given same network conditions, though you cover most browser versions.

Would you say that all things considered, Webp is a sensible default then? And that maybe we don't need to outline it here (in the example)?

KaziMahbuburRahman replied:
Thanks for the feedback. Given the CPU overhead from AVIF encoding and the potential for storage duplication (worst case), WebP-only is a sensible default for most projects. It avoids those costs while still covering most browsers.

I've removed the formats configuration from the example to keep it aligned with the recommended default. The docs already show the AVIF+WebP option for teams that want to optimize for it, so the example doesn't need to demonstrate it.

Commit:
Remove AVIF and WebP formats from image configuration in next.config.js

Commit:
Merge branch 'docs/add-avif-webp-formats-example' of https://github.com/KaziMahbuburRahman/next.js into docs/add-avif-webp-formats-example

icyJoseph committed:
Merge branch 'canary' into docs/add-avif-webp-formats-example
